source/FreeRTOS_IP_Utils.c

```diff
@@ -550,7 +554,7 @@ static void prvChecksumProtocolSetChecksum( BaseType_t xOutgoingPacket,
         *( pxSet->pusChecksum ) = pxSet->usChecksum;
     }
 
-    #if ( ipconfigHAS_DEBUG_PRINTF != 0 )
+    #if ( ipconfigHAS_DEBUG_PRINTF == 0 )
```

Comment:
Instead of this in line 563 can we print already calculated checksum and remove line 562.
Issue is with calling line 562 calling usGenerateProtocolChecksum again, which is messing the return status.